com.unity.formats.fbx/Editor/ExportModelSettings.cs

```diff
@@ -190,7 +232,33 @@ internal class ExportModelSettingsSerialize : ExportOptionsSettingsSerializeBase
         public override bool ExportUnrendered { get { return exportUnrendered; } }
         public void SetExportUnredererd(bool exportUnrendered){ this.exportUnrendered = exportUnrendered; }
         public override bool PreserveImportSettings { get { return preserveImportSettings; } }
-        public void SetPreserveImportSettings(bool preserveImportSettings){ this.preserveImportSettings = preserveImportSettings && !ExportSettings.instance.ExportOutsideProject; }
+        public void SetPreserveImportSettings(bool preserveImportSettings){ this.preserveImportSettings = preserveImportSettings; }
         public override bool AllowSceneModification { get { return false; } }
+
+        public override bool Equals(object e)
+        {
+            var expOptions = e as ExportModelSettingsSerialize;
+            if (expOptions == null)
+            {
+                return false;
+            }
+            return base.Equals(e) && 
+                include == expOptions.include &&
+                lodLevel == expOptions.lodLevel &&
+                objectPosition == expOptions.objectPosition &&
+                exportUnrendered == expOptions.exportUnrendered &&
+                preserveImportSettings == expOptions.preserveImportSettings;
+        }
+
+        public override int GetHashCode()
+        {
+            var bitmask =  base.GetHashCode();
+            bitmask = (bitmask << 2) | (int)include;
+            bitmask = (bitmask << 2) | (int)lodLevel;
+            bitmask = (bitmask << 2) | (int)objectPosition;
+            bitmask = (bitmask << 1) | (exportUnrendered ? 1 : 0);
+            bitmask = (bitmask << 1) | (preserveImportSettings ? 1 : 0);
```

Comment:
Probably better to use XOR than OR in case the enums get more values in them.

Reply:
not sure I understand. Wouldn't we still need to shift the bits more?

Reply:
Ideally yes, since your data is so short. If you want to be sure to update the hash function then make a unit test that verifies your assumptions and has comments saying how and why.

But we don't need to depend on that; in general it's normal for hash codes to collide since you've only got 32 bits for the code, but potentially (though not in your case here) you've got way more data than you can fit in there. 

XOR mixes information about both operands, whereas with OR then if either operand is a 1, the other operand doesn't matter. So with XOR if you change any one field on its own, the hash value always changes, although if you change two at the same time then you can get a hash collision (eg. animatedSkinnedMesh off and include=5 versus animatedSkinnedMesh on and include=1).

Some bedtime reading about hash functions (slightly out of date now but still good basic info):
http://burtleburtle.net/bob/hash/index.html